src/types/timezone.jl

```diff
@@ -17,7 +17,13 @@ const THREAD_TZ_CACHES = Vector{Dict{String,Tuple{TimeZone,Class}}}()
     end
     return cache
 end
-@noinline _tz_cache_length_assert() = @assert false "0 < tid <= length(THREAD_TZ_CACHES)"
+@noinline function _tz_cache_length_assert(; tid)
+    msg = "Assertion failed: 0 < tid <= length(THREAD_TZ_CACHES) where " *
+          "tid=$(tid) " *
+          "length(THREAD_TZ_CACHES)=$(length(THREAD_TZ_CACHES)) " *
+          "Threads.nthreads()=$(Threads.nthreads())"
+    throw(ErrorException(msg))
+end
```

Comment:
In what scenario did you see the assertion error?

Reply:
I encountered it in PackageCompiler, when trying to build a system image in which TimeZones was one of the indirect dependencies. Unfortunately I wasn't able to get it down to a minimal reproducer.